components/clp-py-utils/clp_py_utils/s3_utils.py

```
@@ -392,3 +384,149 @@ def _gen_deletion_config(objects_list: List[str]):
+
+    for object_key, object_size in _iter_s3_objects(s3_client, bucket, key_prefix, first_key):
+        if object_key.endswith("/"):
+            # Skip any object that resolves to a directory-like path
+            continue
+
+        # We need to do both < and > checks since they are handled differently. Ideally, we can do
+        # it with a single comparison. However, Python doesn't support three-way comparison.
+        if object_key < next_key:
+            continue
+        if object_key > next_key:
+            raise ValueError(f"Key `{next_key}` doesn't exist in the bucket `{bucket}`.")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)